src/Core/Client/Request.php

```diff
@@ -429,7 +429,12 @@ public function clearHeaders(): self
      */
     public function getUri(): ?string
     {
-        return $this->getHandler().'?'.$this->getQueryString();
+        $queryString = $this->getQueryString();
+        if ('' === $queryString) {
+            return $this->getHandler();
+        }
+
+        return $this->getHandler().'?'.$queryString;
     }
```

Comment:
@mkalkbrenner I'm not entirely sure why we would ever want to return `null` here? The return value is always used as a string.

```php
    public function getUri(): string
    {
        $handler = $this->getHandler() ?? '';
        $queryString = $this->getQueryString();

        if ('' === $queryString) {
            return $handler;
        }

        return $handler.'?'.$queryString;
    }
```

Reply:
The definitions of return values have been added as a mass update. We already discovered wrong types from time to time. I think that string is correct here.